pymc_marketing/mmm/models/components/lagging.py

```diff
@@ -0,0 +1,394 @@
+        return {
+            "adstock_lambda": {
+                "dist": "Beta",
+                "kwargs": {"alpha": 1, "beta": 3},
+            },
+            "adstock_shape": {
+                "dist": "Beta",
+                "kwargs": {"alpha": 1, "beta": 3},
+            },
+        }
+
+    @property
+    def variable_mapping(self) -> Dict[str, str]:
+        """The mapping from saturation function parameters to variables in a model."""
+        return {
+            "lam": "adstock_lambda",
+            "k": "adstock_shape",
+        }
+
+    def apply(self, data: Union[pm.Data, pm.MutableData]) -> pm.Deterministic:
+        """
+        Apply the Weibull CDF adstock component to the given data.
+
+        Parameters:
+        -----------
+        data : Union[pm.Data, pm.MutableData]
+            The data to apply the adstock component to.
+
+        Returns:
+        --------
+        pm.Deterministic
+            The adstocked data as a PyMC Deterministic variable.
+
+        """
+        self.adstock_lambda_dist = _get_distribution(
+            dist=self.model_config["adstock_lambda"]
+        )
+
+        self.adstock_shape_dist = _get_distribution(
+            dist=self.model_config["adstock_shape"]
+        )
+
+        with pm.modelcontext(self.model):
+            adstock_lambda = self.adstock_lambda_dist(
+                name="adstock_lambda",
+                dims="channel",
+                **self.model_config["adstock_lambda"]["kwargs"],
+            )
+
+            adstock_shape = self.adstock_shape_dist(
+                name="adstock_shape",
+                dims="channel",
+                **self.model_config["adstock_shape"]["kwargs"],
+            )
+
+            return weibull_adstock(
+                x=data,
+                lam=adstock_lambda,
+                k=adstock_shape,
+                type="CDF",
+                l_max=self.max_lagging,
+            )
+
+
+def _get_lagging_function(
+    name: str,
+    max_lagging: int = 4,
+    model: Optional[pm.Model] = None,
+    model_config: Optional[Dict] = None,
+):
+    """
+    Get the lagging function based on the given name.
+
+    Parameters:
+    -----------
+    name : str
+        The name of the lagging function.
+    max_lagging : int
+        The maximum lagging value for the adstock component.
+    model : pm.Model
+        The PyMC model object.
+    model_config : Optional[Dict], optional
+        The configuration dictionary for the model, by default None.
+
+    Returns:
+    --------
+    BaseFunction
+        The lagging function object.
+
+    Raises:
+    -------
+    ValueError
+        If the lagging function name is not recognized.
+
+    """
+    lagging_functions = {
+        "geometric": GeometricAdstockComponent,
+        "weibull_pdf": WeibullPDFAdstockComponent,
+        "weibull_cdf": WeibullCDFAdstockComponent,
+        # Add other lagging function classes here as needed
```

Comment:
Users don't have access to this and is major reason that I am for passing the classes themself. 
Alternatives could 
1. to have this dictionary outside of function allow editing for user to add their own